docs/lifestyles.md

```diff
@@ -47,8 +47,7 @@ This is how you register a component as per web request:
 ```csharp
 Container.Register(Component.For<MyPerWebRequestComponent>().LifestylePerWebRequest());
 ```
-
-:warning: **Registering `PerWebRequestLifestyleModule`:** In order to function properly per web request requires a `IHttpModule` - `Castle.MicroKernel.Lifestyle.PerWebRequestLifestyleModule` to be registered in your web.config:
+:warning: **Registering `PerWebRequestLifestyleModule`:** (Only required if you are **NOT** running version 3.3 or above) In order to function properly per web request requires a `IHttpModule` - `Castle.MicroKernel.Lifestyle.PerWebRequestLifestyleModule` to be registered in your web.config:
```

Comment:
I think it would be better to refer to the dependance on `Microsoft.Web.Infrastructure` for this to happen automatically as per #133.

This functionality also arrived in 3.0.0.

Reply:
I'm not very sure if it arrived in 3.0.0. Going through release notes, it looked like to come in [version 3.3.0 ](https://github.com/castleproject/Windsor/releases/tag/v3.3). Please refer [issue#45](https://github.com/castleproject/Windsor/issues/45) which was added as improvement in version 3.3.0. 

Additionally, should I also mention that it will work without any configuration changes from .NET Framework 4.5 and above?

Reply:
7a08d1a39b2828aa90f97956fbb8f7504ce272dc added the `PreApplicationStartMethod` attribute using the new `PerWebRequestLifestyleModuleRegistration` class for 3.0.0. 86d397c59fe467120afb45b8526b2a44f5c6f4d1 (from pull request #48) changed the reflection code we had that was using internals of `Microsoft.Web.Infrastructure` to use `HttpApplication.RegisterModule` under .NET 4.5, and the previous reflection code under .NET 3.5/4.0. Neither of us were totally right, from Windsor 3.0.0 it would be automatic but how it was automatic changed in 3.3.0 for .NET 4.5.

Windsor 4.x dropped support for .NET 3.5/4.0, however I guess people are still using the 3.x versions.

It looks like the exception Windsor throws if you haven't registered the module is a little outdated, i.e. referring to MVC3 and WebMatrix. I'm not sure how you obtain the M.W.I DLL now, however we don't even need that doco anymore since we only use `HttpApplication.RegisterModule` now since we only support .NET 4.5.

https://github.com/castleproject/Windsor/blob/e92d65fd8bce80c7617f3cd327552177cca9c50b/src/Castle.Windsor/MicroKernel/Lifestyle/PerWebRequestLifestyleModule.cs#L83-L107

Since you are actually using this functionality and obviously saw our docs are wrong I'm going to leave it up to you to decide what you think is worthwhile writing, however I personally see little value in mentioning how to use Windsor on .NET 3.5/4.0 in our docs because no one should be writing new code for those runtimes. It would be great if you could update the exception message in `PerWebRequestLifestyleModule` too.

Thanks

Reply:
I did not get any error message, the reason I got confused was because documentation said following:

> In order to function properly per web request requires a IHttpModule - Castle.MicroKernel.Lifestyle.PerWebRequestLifestyleModule to be registered in your web.config

It did not talk about **what and how error** would come, hence I assumed that it may lead to some unexpected error if I do not add this line.

I understand and absolutely agree with you we that docs should be updated for the latest version, hence we should remove this warning altogether. I will update this PR, removing the warning.